Automation/S3/delete_s3_buckets.py

```diff
@@ -0,0 +1,194 @@
+        print(Fore.GREEN + "Bucket list retrieved! Cycling through list...\n")
+
+        buckets_to_delete = []
+        for bucket in list_buckets_response['Buckets']:
+            bucket_name = bucket['Name']
+            
+            # Retrieve Bucket Information
+            print(f'Retrieving bucket: {bucket_name} information...')
+            bucket_owner_email = get_owner_email(s3_client, bucket_name)
+            bucket_object_list = get_bucket_object_list(s3_client, bucket_name)
+
+            # Display that information
+            print(Fore.GREEN + f'{"Bucket:":<{COL_SIZE}}{bucket_name}')
+            print(Fore.GREEN + f'{"OwnerEmail:":<{COL_SIZE}}{bucket_owner_email}')
+            bucket_object_list_prepared_for_deletion = []
+            if len(bucket_object_list) != 0:
+                print(Fore.GREEN + "Bucket Objects:")
+                for bucket_object in bucket_object_list:
+                    # Prepare just in case we want to delete the bucket
+                    bucket_object_list_prepared_for_deletion.append({
+                        'Key': bucket_object
+                    })
+                    print(bucket_object)
+            
+            # Prompt user for response
+            delete_bucket_response = ""
+            while delete_bucket_response == "":
+                deletion_prompt_response = input(Fore.YELLOW + "Delete bucket: (y/N/exit)")
+                if deletion_prompt_response == "y":
+                    delete_bucket_response = "True"
+                elif deletion_prompt_response == "N":
+                    delete_bucket_response = "False"
+                elif deletion_prompt_response == "exit":
+                    delete_bucket_response = "Exit"
+
+            
+            if delete_bucket_response == "True":
+                deletion_information = {
+                    'BucketName': bucket_name,
+                    'Objects': bucket_object_list_prepared_for_deletion
+                }
+                buckets_to_delete.append(deletion_information)
+                print(Fore.GREEN + "Object scheduled for deletion!")
+            elif delete_bucket_response == "Exit":
+                break
+
+            print()
+        
+        print()
+        if len(buckets_to_delete) > 0:
+            print("Attempting to delete buckets scheduled for deletion...")
+            delete_buckets(s3_client, buckets_to_delete)
+            print(Fore.GREEN + "Buckets deleted! Exiting script.")
+        else:
+            print(Fore.GREEN + "No buckets scheduled for deletion. Exciting script")
+        
+
+    except (ClientError, NoCredentialsError)  as e:
+        print(Fore.RED + f'Error connecting to S3! {e}')
+    
+
+def get_owner_email(s3_client, bucket_name):
+    '''
+    Gets the owner's email of an S3 bucket if that information is included in the bucket tags as OwnerEmail
+
+        Parameters:
+            s3_client (S3Client): the client necessary to execute S3 commands
```

Comment:
Take a look at the following guide: https://realpython.com/documenting-python-code/ it has good suggestions for comments and documenting your code